chainercv/links/model/ssd/multibox_loss.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import division
+
+import numpy as np
+
+import chainer
+import chainer.functions as F
+
+
+def _elementwise_softmax_cross_entropy(x, t):
+    assert x.shape[:-1] == t.shape
+    shape = t.shape
+    x = F.reshape(x, (-1, x.shape[-1]))
+    t = F.flatten(t)
+    return F.reshape(
+        F.softmax_cross_entropy(x, t, reduce='no'), shape)
+
+
+def _hard_negative(x, positive, k):
+    xp = chainer.cuda.get_array_module(x, positive)
+    x = chainer.cuda.to_cpu(x)
+    positive = chainer.cuda.to_cpu(positive)
+    rank = (x * (positive - 1)).argsort(axis=1).argsort(axis=1)
+    hard_negative = rank < (positive.sum(axis=1) * k)[:, np.newaxis]
+    return xp.array(hard_negative)
+
+
+def multibox_loss(mb_locs, mb_confs, gt_mb_locs, gt_mb_labels, k):
+    """Computes multibox losses
+
+    This is a loss function used in [#]_.
```

Comment:
How about

The formulas for the losses can be found in equation (2) and (3) in [#]_.